src/Service.GraphQLBuilder/GraphQLStoredProcedureBuilder.cs

```diff
@@ -13,21 +13,49 @@ namespace Azure.DataApiBuilder.Service.GraphQLBuilder
     public static class GraphQLStoredProcedureBuilder
     {
         /// <summary>
-        /// Helper function to create StoredProcedure Schema for GraphQL.
-        /// It uses the parameters to build the arguments and returns a list
-        /// of the StoredProcedure GraphQL object.
+        /// Adds the stored procedure schema for Query and Mutation types to the provided lists.
         /// </summary>
-        /// <param name="name">Name used for InputValueDefinition name.</param>
+        /// <param name="name">Name of the stored procedure.</param>
         /// <param name="entity">Entity's runtime config metadata.</param>
         /// <param name="dbObject">Stored procedure database schema metadata.</param>
-        /// <param name="rolesAllowed">Role authorization metadata.</param>
-        /// <returns>Stored procedure mutation field.</returns>
-        public static FieldDefinitionNode GenerateStoredProcedureSchema(
+        /// <param name="definitionNodes">List of definition nodes to be modified.</param>
+        /// <param name="fieldDefinitionNodes">List of field definition nodes to be modified.</param>
+        /// <param name="rolesAllowed">Role authorization metadata (optional).</param>
+        public static void AppendStoredProcedureSchema(
             NameNode name,
             Entity entity,
             DatabaseObject dbObject,
+            IList<IDefinitionNode> definitionNodes,
+            IList<FieldDefinitionNode> fieldDefinitionNodes,
             IEnumerable<string>? rolesAllowed = null
-            )
+        )
+        {
+            var storedProcedureDefinition = (StoredProcedureDefinition)dbObject.SourceDefinition;
```

Comment:
nit: our code styling expects using the type name on the left hand side of the equals sign instead of `var`:
StoredProcedureDefinition storedProcedureDefinition